tests/test_sampling_rates.py

```diff
@@ -148,21 +142,18 @@ def validator(data, root_span):
                     f"Message: {data['log_filename']}:"
                     "Metric _sampling_priority_v1 should be set on traces that with sampling decision"
                 )
-            if sampling_priority not in (
-                expected := get_sampling_decision(
-                    context.tracer_sampling_rate, root_span["trace_id"], root_span["meta"]
-                )
-            ):
+
```

Comment:
As sample_from_rate has been simplified, I think it will be better to remove it. Also could you adjust namings and comments to make the test logic crystal clear, something like :

```
    MAX_TRACE_ID = 2 ** 64
    KNUTH_FACTOR = 1111111111111111111

    # details about this logic can be found in 
    # https://github.com/DataDog/architecture/blob/master/rfcs/apm/integrations/priority-sampling/rfc.md
    should_be_kept = ((trace_id * KNUTH_FACTOR) % MAX_TRACE_ID) <= (sampling_rate * MAX_TRACE_ID):

    # sampling priority with strictly positive values means that the trace is kept
    actually_kept = sampling_priority > 0

   if actually_kept is not should_be_kept:
       raise ValueError(
                    f"Trace id {root_span['trace_id']} "
                    f"sampling priority is {sampling_priority}, which is not one of the exepected value"

```

Reply:
Thanks! I proceeded with the improved naming.
I kept `sample_from_rate` as I'll reuse it on a later PR https://github.com/DataDog/system-tests/pull/1476


Note that this next PR will add much more thorough testing of the sampling logic, which will allow us to remove the existing tests.